src/pages/cart.tsx

```diff
@@ -0,0 +1,237 @@
+import React,{ FC, useEffect, useRef, useState } from "react";
+import Popup from "../components/Popup";
+import { getCartList, addCart } from "../scripts/addCart";
+import { useLocation } from "react-router";
+import { Link } from 'react-router-dom';
+import { setUrl, getParam } from "../scripts/setUrl";
+import Product from './product';
+interface product {
```

Comment:
Move this interface in a separate file. You can create types.ts and put it there.